### PR TITLE
core: update controller-runtime version to v0.11.2

### PR DESCRIPTION
**Pull request by subhamkrai**
update controller-runtime version to v0.11.2

Signed-off-by: subhamkrai <srai@redhat.com>

**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: If this is only a documentation change, add the label `skip-ci` on the PR.
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/development-flow.html#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

**Comment by subhamkrai** [comment]
looks the error is due to `kube-object-storage/lib-bucket-provisioner` package using some method from the older version of `klog` which is removed

**Comment by subhamkrai** [comment]
this 
```// InfoLogger provides compatibility with code that relies on the v0.1.0 interface
// Deprecated: use Logger instead. This will be removed in a future release.
type InfoLogger = Logger
```

**Comment by subhamkrai** [comment]
I'll split this into multiple PR as this has lot of other changes too.

**Comment by subhamkrai** [comment]
on top of #10094

File: go.mod
Comment (subhamkrai) [inline go.mod:221]:
```suggestion
	github.com/kube-object-storage/lib-bucket-provisioner => github.com/subhamkrai/lib-bucket-provisioner v0.0.0-20220413062355-7be98bff5b73
```
will rever this once this get merged https://github.com/kube-object-storage/lib-bucket-provisioner/pull/221

**Review by leseb — COMMENTED** [review]
LGTM once the lib-bucket-prov PR merges.

File: go.mod
Comment (travisn) [reply go.mod:221]: Ready to revert? After that, looks good to merge

**Review by travisn — CHANGES_REQUESTED** [review]

File: go.mod
Comment (subhamkrai) [reply go.mod]: done

**Review by travisn — APPROVED** [review]

**Pull request merged**
